### ISSUE TITLE
[Enhancement] detect missing To/Should method calls consistently

### ISSUE THREAD

**Issue by pohly** [nunnatsa/ginkgolinter#216]
**Is your feature request related to a problem? Please describe.**

Suppose there is a wrapper around `gomega.Expect` or any other `gomega` method which returns a `gomega.Assertion` or a similar wrapper for `gomega.Eventually/Consistently` which returns a `gomega.AsyncAssertion`.

Correct:
* `ktesting.Eventually(...).WithTimeout(30 * time.Second).Should(...)`
* `tCtx.Assert(...).Should(...)`
* `tCtx.Require(...).To(...)`
* `tCtx.Expect(...).To(...)`
* `gomega.Expect(...).To(...)`
* `gomega.Eventually(...).Should(...)`

Incorrect:
* `ktesting.Eventually(...).WithTimeout(30 * time.Second)`
* `ktesting.Eventually(...)`
* `tCtx.Assert(...)``
* `tCtx.Require(...)`
* `tCtx.Expect(...)`
* `gomega.Expect(...)`
* `gomega.Eventually(...)`

I think only the last example is currently reported.

**Describe the solution you'd like**

I suspect the detection is currently tied to checking `gomega.Eventually`. It would have to be extended to all functions or methods which return `gomega.Assertion` or `gomega.AsyncAssertion`. If any of those are called without doing anything with the result, then we have a problem.

The following code does not trigger that, but it does trigger another linters "unused variable" (or something like it), so we'd still catch it eventually:
```
func foo() {
    a := gomega.Expect(...)
}
```

**Additional context**

I already implemented `ktesting.Eventually` and `ktesting.TContext.Expect`. I'm currently adding `ktesting.TContext.Require` (an alias) and `ktesting.TContext.Assert` (same without aborting the test).

**Comment by nunnatsa** [comment]
Thanks @pohly . Thanks for the offer.

Currently, the linter does not support wrappers at all. It will require meaningful refactoring to do that. I you have some time to contribute the code, it would be great.

As for `gomega.Expect(...)` - it should be supported.
